src/main/java/com/iexec/sms/secret/app/TeeTaskRuntimeSecretCount.java

```diff
@@ -16,25 +16,30 @@
 
 package com.iexec.sms.secret.app;
 
-import com.iexec.sms.secret.Secret;
-import lombok.Data;
-import lombok.EqualsAndHashCode;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
 import org.hibernate.annotations.GenericGenerator;
 
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
 import javax.persistence.Id;
 
-@Data
-@EqualsAndHashCode(callSuper = true)
 @NoArgsConstructor
+@AllArgsConstructor
 @Entity
-public class ApplicationRuntimeSecret extends Secret {
-    private long index;
+@Getter
+@Builder
+public class TeeTaskRuntimeSecretCount {
```

Comment:
Do you think we need that abstraction elsewhere (outside of ApplicationRuntimeContext)?